### PR TITLE
refactor(compaction): remove manifest from CompactStatus.

### PR DESCRIPTION
**Pull request by zwang28**
## What's changed and what's your intention?

Before this PR, `CompactStatus` is the single source of truth for LSM tree manifest, which is copied to the new `HummockVersion` after compaction. It causes unnecessary data redundancy and lock contention.

This PR removes manifest from `CompactStatus`. 
- When `get_compact_task`, the latest `HummockVersion` is given, which is the single source of truth of manifest.
- Because current compaction file picking algorithm requires certain order of SSTs inside each level, an additional sorting is added in `get_compact_task`, compared with in-place maintained order before this PR.
- `SstableStat`is replaced with existent `SstableInfo`. (SST id -> compact task id) mapping is maintained in a hash map.

## Checklist

- [x] I have written necessary docs and comments

### CODE REVIEW

File: proto/hummock.proto
Comment (hzxa21) [inline proto/hummock.proto:42]: Btw, we can also change this back to sst id (u64) since the key range information is no longer needed. I am also okay of keeping it as it is since I think we will eventually remove `UncommittedEpoch` in `HummockVersion` after we remove `add_tables` and `get_table_id` RPCs.

**Review by hzxa21 — APPROVED** [review]
Thanks for the refactoring. It simplifies the compaction code quite a bit.

Generally LGTM. Please wait until @soundOfDestiny finishes the review before merging.

File: src/meta/src/hummock/compaction.rs
Comment (soundOfDestiny) [inline src/meta/src/hummock/compaction.rs:220]: Is it possible that use a different identifier here?

**Review by soundOfDestiny — APPROVED** [review]
lgtm

File: src/meta/src/hummock/compaction.rs
Comment (zwang28) [reply src/meta/src/hummock/compaction.rs]: renamed.

File: proto/hummock.proto
Comment (zwang28) [reply proto/hummock.proto:42]: These SstableInfo is also used to filter out SSTables in https://github.com/singularity-data/risingwave/pull/1558. So I'd like to keep it in this PR.

**Pull request merged**
